### PR TITLE
Issue #1369: proxy.config.http.server_max_connections over limit fail

### PR DESCRIPTION
**Pull request by shinrich**

### CODE REVIEW

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/1376/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/1269/ for details.

**Comment by bryancall** [comment]
Please don't create issues if you have a PR for it.

**Comment by shinrich** [comment]
Ok.  Still figuring out the new world order :-)

File: proxy/http/HttpSM.cc [hunk 1]
```diff
@@ -4933,8 +4933,10 @@ HttpSM::do_http_server_open(bool raw)
     // between the statement above and the check below.
     // If this happens, we might go over the max by 1 but this is ok.
     if (sum >= t_state.http_config_param->server_max_connections) {
-      ink_assert(pending_action == nullptr);
-      pending_action = eventProcessor.schedule_in(this, HRTIME_MSECONDS(100));
+      // Eventually may want to have a queue as the origin_max_connection does to allow for a combination
+      // of retries and errors.  But at this point, we are just going to allow the error case.
+      t_state.current.state = HttpTransact::CONNECTION_ERROR;
+      call_transact_and_set_next_state(HttpTransact::HandleResponse);
       httpSessionManager.purge_keepalives();
```
Comment (SolidWallOfCode) [inline proxy/http/HttpSM.cc]: Are you sure you want to do this after `call_transact_and_set_next_state`? Normal is to return the return value of that call.

Comment (shinrich) [inline proxy/http/HttpSM.cc]: This is the order we are currently running in Yahoo production.  The purge_keepalives is just clearing out the global pool, so it should not interfere is the set next state.  This was pre-existing code.  I'm guessing the rationale is that is the system is loaded, we should be shutting down idle connections.  Could do the purge first I suppose.

**Comment by shinrich** [comment]
Addressed Alan's comment.

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/1393/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/1287/ for details.

**Comment by SolidWallOfCode** [comment]
Ship it.

**Pull request merged**